diff --git a/src/compiler/spirv/vtn_cmat.c b/src/compiler/spirv/vtn_cmat.c
--- a/src/compiler/spirv/vtn_cmat.c
+++ b/src/compiler/spirv/vtn_cmat.c
@@ -711,8 +711,7 @@ init_clip_scalars(struct vtn_builder *b,
       uint32_t this_index = index_base + f;
       struct vtn_ssa_value *field = vtn_composite_extract(b, src, &this_index, 1);
       struct vtn_ssa_value *new_field = vtn_create_ssa_value(b, field->type);
-      int nc = glsl_get_vector_elements(field->type);
-      assert(nc <= 1);
+      assert(glsl_get_vector_elements(field->type) <= 1);
       new_field->def = vtn_get_nir_ssa(b, w[4 + f]);
 
       new = vtn_composite_insert(b, new, src_type, new_field, &this_index, 1);
-- 
        vtn_push_ssa_value(b, w[2], new);
